django/template/backends/django.py

```diff
@@ -75,10 +75,26 @@ def from_string(self, template_code):
         return Template(self.engine.from_string(template_code), self)
 
     def get_template(self, template_name):
+        template_name, _, partial_name = template_name.partition("#")
+
         try:
-            return Template(self.engine.get_template(template_name), self)
+            template = self.engine.get_template(template_name)
         except TemplateDoesNotExist as exc:
             reraise(exc, self)
+        if not partial_name:
+            return Template(template, self)
+
+        extra_data = getattr(template, "extra_data")
+        partial_contents = extra_data.get("template-partials", {})
+
+        try:
+            partial = partial_contents[partial_name]
```

Comment:
Is there a change that `partial_contents` is None or any other non-mapping type? If yes, we need a test for those cases and also to catch `TypeError` (since `None["foo"]` raises `TypeError` and not `KeyError`).